rfcs/2021-05-23-introduction.md

```diff
@@ -0,0 +1,36 @@
+# SMP agent introduction
+
+## Problem
+
+Allow one agent client connect two connections it has directly, acting as an out-of-band channel.
+
+It can be used both separately, as part of some client functionality, and as part of group protocol.
+
+## Solution
+
+A protocol with commands and message envelopes to exchange the information between parties to establish connection.
+
+### Commands and messages
+
+Below commands are for the scenario when A introduces B to M.
+
+- command `cIdAB INEW cIdAM infoM` - introduce connection cIdB to connection cIdM (response is `OK`)
+- message `cIdBA ICONF invId infoM` - notification to confirm introduction
+- command `cIdBM? ILET invId` - accept offer to be introduced (response is `cIdBM OK`, followed by `ICON`)
+- message `cIdBM ICON invId` - confirmation that connection is established to both introduced parties
+- message `cIdAB IDONE cIdAM` - confirmation that connection is established to the introducer
+
+### Agent envelopes
+
+- `I_NEW extIntroIdM  infoM` - new introduction offered by introducer
+- `I_INV extIntroIdB idBMInv infoB` - invitation to join connection from B to M sent via A
+- `I_NEW extIntroIdB idBMInv infoB` - new introduction forwarded by introducer
```

Comment:
extIntroIdM? that's how it is on the diagram
Is structure of commands described anywhere? It's not completely clear, at least for me
I_NEW X-id idBMInv infoB:
I_NEW - agents command
extIntroIdB - ? 
idBMInv - conn alias that B has for M (but with inv, not sure what that means)
infoB - some information (profile?) about B